src/scripts/developer.ts

```diff
@@ -68,3 +68,104 @@ const resetForm = (developer: Developer) => {
   developer.experience = 0;
   developer.email = "";
 };
+
+//Add Property of developer object
+export const addProperty = <T>(developer: Developer, additionalProps?: T) => {
+  if (additionalProps) {
+    Object.assign(developer, { additionalProperties: additionalProps });
+  }
+  return {
+    ...developer,
+    additionalProperties: additionalProps,
+  };
+};
+
+// Update Developer's Skill
+export const updateSkill = (
+  developers: Developers,
+  developer: Developer,
+  oldSkill: string,
+  newSkill: string
+) => {
+  const currentDeveloper = developers?.find(
+    ({ email }) => email === developer.email
+  );
+
+  if (currentDeveloper) {
+    const index = currentDeveloper?.skills?.indexOf(oldSkill);
+
+    if (index > -1) {
+      currentDeveloper.skills.splice(index, 1, newSkill);
+      developers.splice(
+        developers.findIndex((item) => item === developer),
+        1,
+        currentDeveloper
+      );
+    }
+  }
+};
+
+// Condition for removing a Developer
+export const conditionForRemoving = (developer: Developer) =>
+  developer.age < 18;
+
+// Remove Developer By Condition
+export const removeDeveloperByCondition = (
+  developers: Developers,
+  conditionForRemoving: (developer: Developer) => boolean
+) => {
+  const devToBeRemoved = developers.filter(conditionForRemoving);
+  devToBeRemoved.forEach((item) =>
+    developers.splice(developers.indexOf(item), 1)
+  );
+};
+
+// Sort Developer by Employment and Age
+export const sortDeveloperByEmploymentAndAge = (developers: Developers) => {
+  console.log(
+    developers.sort((a, b) => {
+      const statusOrder = {
+        true: 1,
+        false: 2,
+      };
+      const statusComparison =
+        statusOrder[`${a.isEmployed}`] - statusOrder[`${b.isEmployed}`];
+      if (statusComparison !== 0) {
+        return statusComparison;
+      }
+
+      return a.age - b.age;
+    })
+  );
+};
+
+// list Skills
+export const listSkills = (developers: Developers, developer: Developer) => {
+  const currentDeveloper = developers?.find(
+    ({ email }) => email === developer.email
+  );
+  currentDeveloper?.skills.forEach((skill) => console.log(skill));
+};
+
+//find developers by skills
+export const findDevelopersBySkill = (developers: Developers, skill: string) => {
+   const filterdArrr= developers.filter(item=> item.skills.indexOf(skill) !== -1)
+    console.log(filterdArrr);
+};
+
+// Update Developer 
+  export const updateDeveloper = (developer: Developer, updates: Partial<Developer>) => {
+    for (const key in updates) {
+        if (updates.hasOwnProperty(key) && key in developer) {
```

Comment:
`updates.hasOwnProperty(key)` why checking this?

Reply:
I have used hasOwnProperty because if the updates object that we are passing is inherited from another object and have inherited properties but I only want to check the direct properties of updates object. For e.g.-

const parentObj= { age: 35 }; 
const updates = Object.create(parentObj); 
updates.name = "Raj"; // Direct property
const developer = { name: "John", age: 30 }; // Developer object

In above case , we only wanted to change the name but due to inherited properties, name will also change.
So, to keep a strict check, I have used hasOwnProperty because it checks only for direct properties.